cli/command/container/create.go

```diff
@@ -193,14 +204,26 @@ func createContainer(ctx context.Context, dockerCli command.Cli, containerConfig
 		}
 	}
 
-	//create the container
-	response, err := dockerCli.Client().ContainerCreate(ctx, config, hostConfig, networkingConfig, opts.name)
+	//create the container, pulling the image (or not) based on opts.pull
+	var response container.ContainerCreateCreatedBody
+
+	if opts.pull == PullImageAlways {
+		if err := pullImage(ctx, dockerCli, config.Image, opts.platform, stderr); err != nil {
```

Comment:
Would it be helpful to put all this behavior into a closure that can be reused in the error handling for `ContainerCreate`?

Reply:
Yup that's cleaner.